packages/govuk-frontend/src/govuk/components/header/header.mjs

```diff
@@ -44,20 +46,14 @@ export class Header {
     this.$menu = this.$menuButton && $module.querySelector(
       `#${this.$menuButton.getAttribute('aria-controls')}`
     )
-  }
 
-  /**
-   * Initialise component
-   *
-   * Check for the presence of the header, menu and menu button – if any are
-   * missing then there's nothing to do so return early.
-   * Apply a matchMedia for desktop which will trigger a state sync if the browser
-   * viewport moves between states.
-   */
-  init () {
-    // Check that required elements are present
-    if (!this.$module || !this.$menuButton || !this.$menu) {
-      return
+    if (
+      !(
+        this.$menuButton instanceof HTMLButtonElement ||
```

Comment:
We tend to not be specific with these checks so `HTMLElement` is good enough usually, but if we need the compiler to guard specific browser APIs then go for it

For example:

**Character count** checks for `HTMLTextAreaElement` or `HTMLInputElement`
**Radios** and **Checkboxes** check for `HTMLInputElement`
**Skip link** checks for `HTMLAnchorElement`

```suggestion
        this.$menuButton instanceof HTMLElement ||
```